astrolabe/runner.py

```diff
@@ -187,8 +187,26 @@ def run(self, persist_cluster=False, startup_time=1):
                 LOGGER.info("Cluster maintenance complete")
                 
             elif op_name == 'testFailover':
-                self.cluster_url['restartPrimaries'].post()
-                
+                timer = Timer()
+                timer.start()
+                timeout = 90
+
+                # DRIVERS-1585: failover may fail due to the cluster not being
+                # ready. Retry failover up to a timeout if the
+                # CLUSTER_RESTART_INVALID error is returned from the call
+                while True:
```

Comment:
It doesn't change functionality but I would write this a bit differently to make the intent clear:

```
                while True:
                    try:
                        self.cluster_url['restartPrimaries'].post()
                    except AtlasApiError as exc:
                        if exc.error_code != 'CLUSTER_RESTART_INVALID':
                            raise
                    else:
                        break

                    if timer.elapsed > timeout:
                        raise PollingTimeoutError("Could not test failover as cluster wasn't ready")
                    else:
                        sleep(5)
```

Reply:
Thanks for the suggestion, I wasn't aware `try...except...else` was a thing. TIL :)